patches/minecraft/net/minecraft/entity/passive/HorseArmorType.java.patch

```diff
@@ -0,0 +1,36 @@
+--- ../src-base/minecraft/net/minecraft/entity/passive/HorseArmorType.java
++++ ../src-work/minecraft/net/minecraft/entity/passive/HorseArmorType.java
+@@ -62,16 +62,33 @@
+ 
+     public static HorseArmorType func_188580_a(@Nullable ItemStack p_188580_0_)
+     {
++        if (p_188580_0_ != null && p_188580_0_.func_77973_b() instanceof net.minecraftforge.common.IHorseArmor) return ((net.minecraftforge.common.IHorseArmor) p_188580_0_.func_77973_b()).getArmorType(p_188580_0_);
+         return p_188580_0_ == null ? NONE : func_188576_a(p_188580_0_.func_77973_b());
+     }
+ 
+     public static HorseArmorType func_188576_a(@Nullable Item p_188576_0_)
+     {
++        if (p_188576_0_ instanceof net.minecraftforge.common.IHorseArmor) return ((net.minecraftforge.common.IHorseArmor) p_188576_0_).getArmorType(new ItemStack(p_188576_0_));
+         return p_188576_0_ == Items.field_151138_bX ? IRON : (p_188576_0_ == Items.field_151136_bY ? GOLD : (p_188576_0_ == Items.field_151125_bZ ? DIAMOND : NONE));
+     }
+ 
+     public static boolean func_188577_b(@Nullable Item p_188577_0_)
+     {
++        if (p_188577_0_ instanceof net.minecraftforge.common.IHorseArmor) return true;
+         return func_188576_a(p_188577_0_) != NONE;
+     }
++    
++    // Forge    
++    private HorseArmorType(int armorStengthIn, String textureName)
```

Comment:
There is nothing in here deprecating the old methods, or killing off the usages of the old bad methods.
The constructor also has no documentation. 
Also by preliminary glance at how this class is used, it looks like you need some mechanic of specifying the domain for the texture in EntityHorse.setHorseTexturePaths